### PR TITLE
test: Run cilium status preflight check

### PR DESCRIPTION
**Pull request by tgraf**
During a test failure, it has been observed that it is difficult to assess the
failure reason if the connectivity health check fails due to a timeout without
having cilium status output in the command log. Thus add cilium status output
to the command log with health information all nodes included.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6594)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1726,6 +1730,30 @@ func (kub *Kubectl) CiliumPreFlightCheck() error {
...
+		return err
+	}
+
+	for _, pod := range ciliumPods {
+		status := kub.CiliumExec(pod, "cilium status --all-health --all-nodes")
+		if !status.WasSuccessful() {
+			return fmt.Errorf("cilium-agent '%s': Cannot run cilium status: %s",
+				pod, status.OutputPrettyPrint())
+		}
+		if status.GetExitCode() != 0 {
```
Comment (eloycoto) [inline test/helpers/kubectl.go]: AFAIK the WasSuccessful is set if the status is 0, so these three lines does not make sense at all.

**Review by eloycoto — COMMENTED** [review]
I'm ok with the change, but I'm a bit worried about this one, that check should be in the livenessProbe and fail in there no, how a pod can be ready and the status is not correct?

**Comment by tgraf** [comment]
Worried about what? The commit message explains that this is about providing status visibility while we wait for pre-flight conditions to become true.

As discussed in #development, on health issues, the only output in the logs is:
```
stderr:
Error: Cannot get status/probe: context deadline exceeded
```

No further context besides seeing the number of controllers failing. This adds overall status visibility which includes connectivity health for each node including the information whether the health daemon is up and running.

**Comment by tgraf** [comment]
test-me-please

Hit #6600

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**